framework/src/st_things/things_stack/cloud/cloud_manager.c

```diff
@@ -1311,7 +1312,7 @@ OCStackResult log_in_out_to_cloud(bool value, things_timeout_s *timeout)
 	OCStackResult res = OC_STACK_ERROR;
 	OCClientResponseHandler callback = NULL;
 	things_check_time_out_call_func calltimeout = NULL;
-	char *device_id = NULL;
+	const char *device_id = NULL;
```

Comment:
Is this possible?
change a const variable after variable initialize?

I think `const char * device_id = OCGetServerInstanceIDString();` Should not we do like this?

Reply:
It is working fine. I am not changing the value which is pointed by device_id(*device_id). Here pointer itself changed(device_id).